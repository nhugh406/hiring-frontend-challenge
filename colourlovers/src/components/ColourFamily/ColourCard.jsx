import React from 'react';
import { array, string, number } from 'prop-types';
import './styles.scss';

const propTypes = {
  id: number.isRequired,
  title: string.isRequired,
  userName: string.isRequired,
  numViews: number.isRequired,
  numVotes: number.isRequired,
  numComments: number.isRequired,
  numHearts: number.isRequired,
  rank: number.isRequired,
  dateCreated: string.isRequired,
  colors: array.isRequired,
  description: string.isRequired,
  url: string.isRequired,
  imageUrl: string.isRequired,
  badgeUrl: string.isRequired,
  apiUrl: string.isRequired,
};

function ColourCard({
  id,
  title,
  userName,
  numViews,
  numVotes,
  numComments,
  numHearts,
  rank,
  dateCreated,
  colors,
  description,
  url,
  imageUrl,
  badgeUrl,
  apiUrl,
}) {
  return (
    <div key={id} className="card">
      <div className="details">
        <div className="subsection">
          <div className="title">{title}</div>
          <div className="subTitle">
            by {userName} at {dateCreated}
          </div>
        </div>

        <div className="buttonSection">
          <div>{numViews} views</div>
          <div>{numVotes} votes</div>
        </div>
      </div>
      <div className="color-block">
        {colors.map((col) => (
          <div style={{ background: `#${col}` }} key={col}></div>
        ))}
      </div>
    </div>
  );
}

ColourCard.propTypes = propTypes;
export default ColourCard;
