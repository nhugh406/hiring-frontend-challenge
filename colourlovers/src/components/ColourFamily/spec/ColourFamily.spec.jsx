import React from 'react';
import { render } from '@testing-library/react';

import ColourCard from '../ColourCard';

const mockProps = {
  id: 92095,
  title: 'Giant Goldfish',
  userName: 'manekineko',
  numViews: 1087457,
  numVotes: 12107,
  numComments: 777,
  numHearts: 4.5,
  rank: 1,
  dateCreated: '2007-07-03 10:42:02',

  colors: ['69D2E7', 'A7DBD8', 'E0E4CC', 'F38630', 'FA6900'],
  description:
    'I-MOO\r\n<div style="width: 300px; text-align: center;"><a href="http://www.colourlovers.com/contests/moo/minicard/2291466" target="_blank" style="display: block; margin-bottom: 5px; width: 300px; height: 120px; -moz-box-shadow: 0 1px 4px #d1d1d1; -webkit-box-shadow: 0 1px 4px #d1d1d1; box-shadow: 0 1px 4px #d1d1d1; filter: progid:DXImageTransform.Microsoft.Shadow(Strength=1, Direction=180, Color=',
  url: 'http://www.colourlovers.com/palette/92095/Giant_Goldfish',
  imageUrl:
    'http://www.colourlovers.com/paletteImg/69D2E7/A7DBD8/E0E4CC/F38630/FA6900/Giant_Goldfish.png',
  badgeUrl:
    'http://www.colourlovers.com/images/badges/p/92/92095_Giant_Goldfish.png',
  apiUrl: 'http://www.colourlovers.com/api/palette/92095',
};

describe('Colour Family Component Tests', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<ColourCard {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
