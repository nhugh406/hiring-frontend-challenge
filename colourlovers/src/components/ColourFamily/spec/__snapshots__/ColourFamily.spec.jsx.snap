// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colour Family Component Tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="details"
    >
      <div
        class="subsection"
      >
        <div
          class="title"
        >
          Giant Goldfish
        </div>
        <div
          class="subTitle"
        >
          by manekineko at 2007-07-03 10:42:02
        </div>
      </div>
      <div
        class="buttonSection"
      >
        <div>
          1087457 views
        </div>
        <div>
          12107 votes
        </div>
      </div>
    </div>
    <div
      class="colorBlock"
    >
      <div
        style="background: rgb(105, 210, 231);"
      />
      <div
        style="background: rgb(167, 219, 216);"
      />
      <div
        style="background: rgb(224, 228, 204);"
      />
      <div
        style="background: rgb(243, 134, 48);"
      />
      <div
        style="background: rgb(250, 105, 0);"
      />
    </div>
  </div>
</DocumentFragment>
`;
